src/main/java/runwar/Server.java

```diff
@@ -409,19 +408,23 @@ public synchronized void startServer(final ServerOptions options) throws Excepti
         }
 
         if (osName != null && osName.startsWith("Mac OS X")) {
-            Image dockIcon = Tray.getIconImage(dockIconPath);
-            System.setProperty("com.apple.mrj.application.apple.menu.about.name", processName);
-            System.setProperty("com.apple.mrj.application.growbox.intrudes", "false");
-            System.setProperty("apple.laf.useScreenMenuBar", "true");
-            System.setProperty("-Xdock:name", processName);
-            try {
-                Class<?> appClass = Class.forName("com.apple.eawt.Application");
-                Method getAppMethod = appClass.getMethod("getApplication");
-                Object appInstance = getAppMethod.invoke(null);
-                Method dockMethod = appInstance.getClass().getMethod("setDockIconImage", java.awt.Image.class);
-                dockMethod.invoke(appInstance, dockIcon);
-            } catch (Exception e) {
-                LOG.warn("error setting dock icon image", e);
+            if (serverOptions.dockEnable()) {
+                Image dockIcon = Tray.getIconImage(dockIconPath);
+                System.setProperty("com.apple.mrj.application.apple.menu.about.name", processName);
+                System.setProperty("com.apple.mrj.application.growbox.intrudes", "false");
+                System.setProperty("apple.laf.useScreenMenuBar", "true");
+                System.setProperty("-Xdock:name", processName);
+                try {
+                    Class<?> appClass = Class.forName("com.apple.eawt.Application");
+                    Method getAppMethod = appClass.getMethod("getApplication");
+                    Object appInstance = getAppMethod.invoke(null);
+                    Method dockMethod = appInstance.getClass().getMethod("setDockIconImage", java.awt.Image.class);
+                    dockMethod.invoke(appInstance, dockIcon);
+                } catch (Exception e) {
+                    LOG.warn("error setting dock icon image", e);
+                }
+            } else {
+                System.setProperty("apple.awt.UIElement", "true");

```

Comment:
@mgmathus I'm curious, what does this line of code do here when the doc icon is disabled?

Reply:
@bdw429s by default, when a java app is launched on Mac OS, there is a coffee cup icon that pops in the dock, what Denny Did with that portion of code is customize that icon with the one used in the tray icon, so the dock icon is not the generic one,

The dock icon is been removed by line 427

Reply:
Ahh, makes sense.  Thanks for the explanation!